### PR TITLE
TypeScript FrontEnd: Add Account Files

### PR DESCRIPTION
**Pull request by Ferotiq**
### Description

Adds account files, config.ts, and db.ts.

Closes #

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -30,12 +30,12 @@
     "eslint-plugin-promise": "5.1.0",
     "eslint-plugin-require-path-exists": "1.1.9",
     "husky": "4.3.0",
-    "prettier": "2.1.2",
+    "prettier": "^2.5.1",
```
Comment (Bruception) [inline package.json:33]: Save exact

**Comment by Miodec** [comment]
Graph points have timestamp messed up somehow
image

File: frontend/src/scripts/elements/commandline-lists.ts [hunk 2]
```diff
@@ -205,23 +205,23 @@ const commandsTags: MonkeyTypes.CommandsGroup = {
 };
 
 export function updateTagCommands(): void {
-  if (DB.getSnapshot()?.tags?.length > 0) {
+  if (DB.getSnapshot()?.tags?.length || 0 > 0) {
```
Comment (Miodec) [inline frontend/src/scripts/elements/commandline-lists.ts:208]: Whats up with this

File: frontend/src/scripts/elements/commandline-lists.ts [hunk 3]
```diff
@@ -1091,7 +1091,7 @@ const commandsRandomTheme: MonkeyTypes.CommandsGroup = {
       display: "fav",
       configValue: "fav",
       exec: (): void => {
-        UpdateConfig.setRandomTheme("fav");
+        UpdateConfig.setRandomTheme("favorite");
```
Comment (Miodec) [inline frontend/src/scripts/elements/commandline-lists.ts:1094]: This breaks random theme, revert.

File: frontend/src/scripts/elements/commandline-lists.ts (hunk 2 above)
Comment (Ferotiq) [reply frontend/src/scripts/elements/commandline-lists.ts:208]: length can be undefined and undefined can't be compared with a number.

Comment (Miodec) [reply frontend/src/scripts/elements/commandline-lists.ts:208]: OOoooooh, i usually use `??` for this so thats why i got confused.

**Comment by Miodec** [comment]
image

Error when trying to open the commandline.

File: frontend/src/scripts/account/result-filters.ts [hunk 4]
```diff
@@ -104,50 +106,67 @@ export function load() {
...
   filters = defaultResultFilters;
   save();
 }
 
-export function updateActive() {
-  let aboveChartDisplay = {};
-  Object.keys(getFilters()).forEach((group) => {
+type AboveChartDisplay = MonkeyTypes.PartialRecord<
+  MonkeyTypes.Group,
+  { all: boolean; array?: (string | boolean)[] }
```
Comment (Miodec) [inline frontend/src/scripts/account/result-filters.ts:148]: Why array of strings?

**Comment by Miodec** [comment]
Also in the future please please avoid unrelated changes in the PR. This one is about converting account files (and db and config), but there is 41 files changed. Things like adding missing types everywhere should be a separate PR.

File: frontend/src/scripts/popups/edit-tags-popup.ts [hunk 5]
```diff
@@ -91,9 +91,11 @@ async function apply(): Promise<void> {
       Notifications.add(response.data.message);
     } else {
       Notifications.add("Tag added", 1);
-      DB.getSnapshot().tags.push({
+      DB.getSnapshot().tags?.push({
         name: response.data.name,
         _id: response.data._id,
+        personalBests: {} as MonkeyTypes.PersonalBests,
+        active: true,
```
Comment (Miodec) [inline frontend/src/scripts/popups/edit-tags-popup.ts:98]: Remove. Automatically enabling could be confusing IMO.

File: frontend/src/scripts/pages/account.ts [hunk 6]
```diff
@@ -641,6 +652,12 @@ export function update(): void {
         "Words per Minute";
     }
 
+    console.log(
+      chartData.map((a) => {
+        return { x: a.x, ts: a.timestamp };
+      })
+    );
```
Comment (Miodec) [inline frontend/src/scripts/pages/account.ts:659]: Remove

File: frontend/src/scripts/db.ts [hunk 7]
```diff
@@ -26,19 +28,27 @@ export function setSnapshot(newSnapshot) {
...
+  const defaultSnap: MonkeyTypes.Snapshot = {
     results: undefined,
-    personalBests: {},
+    personalBests: {
+      time: { custom: [] },
+      words: { custom: [] },
+      zen: { zen: [] },
+      quote: { custom: [] },
+      custom: { custom: [] },
+    },
```
Comment (Miodec) [inline frontend/src/scripts/db.ts:43]: time, words and quote wont have a `custom` array.

File: frontend/src/scripts/controllers/account-controller.js [hunk 8]
```diff
@@ -134,7 +134,7 @@ export async function getDataAndInit() {
         if (response?.status == 200) {
           nameGood = true;
           Notifications.add("Name updated", 1);
-          DB.getSnapshot().name = name;
+          snapshot.name = name;
```
Comment (Miodec) [inline frontend/src/scripts/controllers/account-controller.js:137]: Is this saved?

**Review by Miodec — APPROVED** [review]

**Pull request merged**